Classes/EventListener/WebpCreator.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netlogix\Nximageoptimizer\EventListener;
+
+use TYPO3\CMS\Core\Resource\Driver\DriverInterface;
+use TYPO3\CMS\Core\Resource\Event\AfterFileProcessingEvent;
+use TYPO3\CMS\Core\Resource\ProcessedFile;
+use TYPO3\CMS\Core\Utility\CommandUtility;
+
+class WebpCreator extends AbstractImageOptimizer
+{
+
+	public function createWebpVersion(AfterFileProcessingEvent $event)
+	{
+		if (!$this->isEnabled($event->getProcessedFile())) {
+			return;
+		}
+
+		$this->createWebpImage($event->getProcessedFile(), $event->getDriver());
+	}
+
+	/**
+	 * Creates a .webp version of the processed image file.
+	 *
+	 * @param ProcessedFile $processedFile
+	 * @param DriverInterface $driver
+	 * @return void
+	 */
+	private function createWebpImage(ProcessedFile $processedFile, DriverInterface $driver): void
+	{
+		if ($this->configuration['disableAutomaticWebpCreation']) {
+			return;
+		}
+
+		$targetFileName = $processedFile->getName() . '.webp';
+		$targetFolder = $processedFile->getParentFolder();
+
+		$localProcessingPath = $processedFile->getForLocalProcessing();
```

Comment:
This will create a temporary file we don't need.
In the code in Line 62-70 we add this temporary file to the driver and remove the currently processed webp file this the original temporary file.

Reply:
See my comment above about localDriver and FAL API